app/javascript/components/upload/UploadWizard.jsx

```diff
@@ -269,49 +270,72 @@ export function RawUploadWizard({ studyAccession, name }) {
 
   /** save the given file and perform an upload if a selected file is present */
   async function saveFile(file) {
+    let fileToSave = file
     let studyFileId = file._id
 
+    // in AnnData case of saving file when you're doing an update
     if (isAnnDataExperience) {
-      // enable ingest of data by setting reference_anndata_file = false
-      file['reference_anndata_file'] = false
-      formState.files.forEach(fileFormData => {
-        if (fileFormData.file_type === 'Cluster') {
-          fileFormData.data_type = 'cluster'
-
-          // mulitple clustering forms are allowed so add each as a fragment to the AnnData file
-          const annDataFile = formState.files.find(f => f.file_type === 'AnnData')
-          annDataFile?.ann_data_file_info ? '': annDataFile['ann_data_file_info'] = {}
-          const fragments = annDataFile.ann_data_file_info?.data_fragments || []
-          fragments.push(fileFormData)
-
-          annDataFile.ann_data_file_info.data_fragments = fragments
+      // necessary check for when updating a Clustering or Expression matrix form the fileToBeSaved needs to be updated to the AnnData file
+      // if the file update is on an existing fragment it will have a data_type rather than file_type
+      // or if it's a new clustering it won't have a data_type but will have an obsm key name field
```

Comment:
Could you break this up with some punctuation?  

The comment strikes me as important, but I've tried reading it a few times, and I unfortunately can't really understand it.  (I don't intend to wordsmith code comments; I'm just seeking to ensure maintainers can clearly understand what's written here.)